packages/realm/src/Realm.ts

```diff
@@ -284,24 +290,67 @@ export class Realm {
   /**
    * Sets the log level.
    * @param level - The log level to be used by the logger. The default value is `info`.
+   * @param category - The category/component to set the log level for. If omitted, log level is set for all known categories.
    * @note The log level can be changed during the lifetime of the application.
    * @since 12.0.0
+   * @example
+   * Realm.setLogLevel({ category: LogCategory.Realm, level: "all" });
    */
-  static setLogLevel(level: LogLevel) {
-    const bindingLoggerLevel = toBindingLoggerLevel(level);
-    binding.Logger.setDefaultLevelThreshold(bindingLoggerLevel);
+  static setLogLevel(_arg: LogLevel | LogArgs) {
+    // It is not possible to overload a static function: https://github.com/microsoft/TypeScript/issues/18945
+    // FIXME: don't use `arguments` but find a proper type
+    if (arguments.length === 1) {
+      // eslint-disable-next-line prefer-rest-params
+      const arg = arguments[0];
+      if (arg.level) {
+        const level = arg.level;
+        if (arg.category) {
+          const category = arg.category;
+          assert(Object.values(LogCategory).includes(category));
+          binding.LogCategoryRef.getCategory(category).setDefaultLevelThreshold(toBindingLoggerLevel(level));
+        } else {
+          Object.values(LogCategory).forEach((category) => {
+            Realm.setLogLevel({ level, category });
+          });
+        }
+      } else {
+        const level = arg;
+        Object.values(LogCategory).forEach((category) => {
+          Realm.setLogLevel({ level, category });
+        });
+      }
+    } else {
+      throw new Error(`Wrong number of arguments - expected 1, got ${arguments.length}`);
+    }
   }
 
   /**
    * Sets the logger callback.
    * @param loggerCallback - The callback invoked by the logger. The default callback uses `console.log`, `console.warn` and `console.error`, depending on the level of the message.
-   * @note The logger callback needs to be setup before opening the first realm.
+   * @note The logger callback needs to be setup before opening the first Realm.
    * @since 12.0.0
+   * @example
+   * Realm.setLogger(({ category, level, message }) => {
+   *   console.log(`[${category} - ${level}] ${message}`);
+   * });
    */
   static setLogger(loggerCallback: LoggerCallback) {
```

Comment:
An alternative here is also to modify the `LoggerCallback` to e.g.:
```ts
export type Logger = (level: NumericLogLevel, message: string, category?: LogCategory) => void;
```

Then this function could be:
```ts
  static setLogger(loggerCallback: LoggerCallback) {
    const logger = binding.Helpers.makeLogger((category, level, message) => {
      loggerCallback(fromBindingLoggerLevelToLogLevel(level), message, category as LogCategory);
    });
    binding.Logger.setDefaultLogger(logger);
  }
```